proofs/eo/cpc/rules/Strings.eo

```diff
@@ -179,6 +179,7 @@
 ;   An equality involving the given child of t. In particular, we infer what
 ;   portion of the prefix (resp. suffix) it is entailed to be based on
 ;   reasoning about adjacent constants.
+; TODO: update usage of $str_overlap
```

Comment:
Can we avoid this?

Reply:
This proof rule is already calling `$str_overlap` but instead can be simplified to use `$str_has_overlap` introduced in this PR.  This will allow us to delete the former method.